cluster-autoscaler/cloudprovider/gke/gke_manager.go

```diff
@@ -498,7 +504,9 @@ func (m *gkeManagerImpl) refreshClusterResources() error {
 	if err != nil {
 		return err
 	}
-	m.refreshNodePools(cluster.NodePools)
+	if err = m.refreshNodePools(cluster.NodePools); err != nil {
```

Comment:
This makes me wonder how we want to handle errors here. I don't think a single broken node group should cause us to abort all operations, but then we probably shouldn't break from the loop in refreshNodePools either. @MaciekPytel WDYT?

Reply:
+1 on that thought. We probably shouldn't react too fiercely on one error. Ideally we should have a simple error budget for the refreshNodePools call. Eg. Greater than 50% throw errors

Reply:
I'm not confident with this change as-is. Since this is a drive-by fix, can we revert it and address the issue in a separate PR?